yazi-core/src/tab/selected.rs

```diff
@@ -0,0 +1,251 @@
+use std::{collections::{BTreeSet, HashMap}, path::PathBuf};
+
+use yazi_shared::fs::Url;
+
+#[derive(Default)]
+pub struct Selected {
+	inner:   BTreeSet<Url>,
+	parents: HashMap<PathBuf, usize>,
+}
+
+impl Selected {
+	pub fn get_inner(&self)->BTreeSet<Url>{
+		return self.inner.clone()
+	}
+	pub fn new() -> Self { Selected { inner: BTreeSet::new(), parents: HashMap::new() } }
+
+	pub fn insert(&mut self, url: Url) -> bool { self.insert_many(&[&url]) }
+
+	pub fn insert_many(&mut self, urls: &[&Url]) -> bool {
+		if urls.is_empty() {
+			return false;
+		}
+		let url_buf = urls[0].to_path_buf();
```

Comment:
I simply changed the `url` in this line to `urls[0]`

Reply:
Good idea!